beacon-chain/db/kv/state.go

```diff
@@ -914,15 +915,41 @@ func (s *Store) CleanUpDirtyStates(ctx context.Context, slotsPerArchivedPoint pr
 				return ctx.Err()
 			}
 
-			finalizedChkpt := bytesutil.ToBytes32(f.Root) == bytesutil.ToBytes32(v)
+			oRoot, err := s.OriginCheckpointBlockRoot(ctx)
```

Comment:
Can you reorganize the stuff up here to only execute before the check statements that require it? That way we avoid doing extra computation - especially in this case where we do an extra db index read for every item we check.

Reply:
Aside from this I was referring to computing the slot etc, which you only need to do after the finalized root comparison. But that stuff is no big deal, this origin checkpoint block root query is the only one I'd really like to see moved until it is needed (line 946).

Reply:
You could also do the OriginCheckpointBlockRoot read in a separate tx before the forEach. The value doesn't change over the life of the db.